californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayerParameters.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Lari Hotari and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Lari Hotari - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.californium.core.network.stack;
+
+import org.eclipse.californium.core.network.config.NetworkConfig;
+
+/**
+ * Contains the configuration values for {@link ReliabilityLayer} and {@link CongestionControlLayer}
+ */
+public class ReliabilityLayerParameters {
+	private final int ackTimeout;
+	private final float ackRandomFactor;
+	private final float ackTimeoutScale;
+	private final int maxRetransmit;
+	private final int nstart;
+
+	ReliabilityLayerParameters(int ackTimeout, float ackRandomFactor, float ackTimeoutScale, int maxRetransmit, int nstart) {
+		this.ackTimeout = ackTimeout;
+		this.ackRandomFactor = ackRandomFactor;
+		this.ackTimeoutScale = ackTimeoutScale;
+		this.maxRetransmit = maxRetransmit;
+		this.nstart = nstart;
+	}
+
+	public int getAckTimeout() {
+		return ackTimeout;
+	}
+
+	public float getAckRandomFactor() {
+		return ackRandomFactor;
+	}
+
+	public float getAckTimeoutScale() {
+		return ackTimeoutScale;
+	}
+
+	public int getMaxRetransmit() {
+		return maxRetransmit;
+	}
+
+	public int getNstart() {
+		return nstart;
+	}
+
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	public static final class Builder {
+		private int ackTimeout;
+		private float ackRandomFactor;
+		private float ackTimeoutScale;
+		private int maxRetransmit;
+		private int nstart;
+
+		private Builder() {
+		}
+
+		public Builder applyConfig(NetworkConfig config) {
+			ackTimeout = config.getInt(NetworkConfig.Keys.ACK_TIMEOUT);
+			ackRandomFactor = config.getFloat(NetworkConfig.Keys.ACK_RANDOM_FACTOR);
+			ackTimeoutScale = config.getFloat(NetworkConfig.Keys.ACK_TIMEOUT_SCALE);
+			maxRetransmit = config.getInt(NetworkConfig.Keys.MAX_RETRANSMIT);
+			nstart = config.getInt(NetworkConfig.Keys.NSTART);
+			return this;
+		}
```

Comment:
I still not like so much the idea to make this class dependent of NetworkConfig but this time there is not easy alternative :/